README.md

```diff
@@ -292,6 +292,142 @@ app.register(GQL, {
 })
 ```
 
+### Federation metadata support
+
+The signature of the method is the same as a standard resolver: `__resolveReference(source, args, context, info)` where the `source` will contain the reference object that needs to be resolved.
+
+```js
+'use strict'
+
+const Fastify = require('fastify')
+const GQL = require('fastify-gql')
+
+const users = {
+  1: {
+    id: '1',
+    name: 'John',
+    username: '@john'
+  },
+  2: {
+    id: '2',
+    name: 'Jane',
+    username: '@jane'
+  }
+}
+
+const app = Fastify()
+const schema = `
+  extend type Query {
+    me: User
+  }
+
+  type User @key(fields: "id") {
+    id: ID!
+    name: String
+    username: String
+  }
+`
+
+const resolvers = {
+  Query: {
+    me: () => {
+      return users['1']
+    }
+  },
+  User: {
+    __resolveReference: (source, args, context, info) => {
+      return users[source.id]
+    }
+  }
+}
+
+app.register(GQL, {
+  schema,
+  resolvers,
+  federationMetadata: true
+})
+
+app.get('/', async function (req, reply) {
+  const query = '{ _service { sdl } }'
+  return app.graphql(query)
+})
+
+app.listen(3000)
+```
+
+### Federation with __resolveReference caching
+
+Just like standard resolvers, the `__resolveReference` resolver can be a performance bottleneck. To avoid this, the it is strongly recommended to define the `__resolveReference` function for an entity as a [loader](#defineLoaders).
+
+```js
+'use strict'
+
+const Fastify = require('fastify')
+const GQL = require('fastify-gql')
+
+const users = {
+  1: {
+    id: '1',
+    name: 'John',
+    username: '@john'
+  },
+  2: {
+    id: '2',
+    name: 'Jane',
+    username: '@jane'
+  }
+}
+
+const app = Fastify()
+const schema = `
+  extend type Query {
+    me: User
+  }
+
+  type User @key(fields: "id") {
+    id: ID!
+    name: String
+    username: String
+  }
+`
+
+const resolvers = {
+  Query: {
+    me: () => {
+      return users['1']
+    }
+  }
+}
+
+const loaders = {
+  User: {
+    __resolveReference: {
+      async loader(queries, context) {
+        // This should be a bulk query to the database
+        return queries.map(({ obj }) => users[obj.id])
+      },
+      opts: {
+        cache: true
+      }
```

Comment:
I don't think we support `cache: true` at all here.

Reply:
It is passed to the factory https://github.com/mcollina/fastify-gql/blob/ca23bb6c22d88b021a4b4503cfae321895e5dc8d/index.js#L186 but it's the same as not setting the `opts` property at all and passing only the loader function.